src/test/java/no/unit/nva/model/util/PublicationGenerator.java

```diff
@@ -410,220 +421,260 @@ private static Publication generatePublicationWithReference(Reference reference)
         return new Book.BookBuilder()
-                   .withIsbnList(List.of("9780201309515"))
-                   .withPublisher(new UnconfirmedPublisher("My publisher dot com"))
-                   .withSeriesNumber("123")
-                   .withSeries(new UnconfirmedSeries("Explorations in ego"))
-                   .build();
+            .withIsbnList(List.of("9780201309515"))
+            .withPublisher(new UnconfirmedPublisher("My publisher dot com"))
+            .withSeriesNumber("123")
+            .withSeries(UnconfirmedSeries.fromTitle("Explorations in ego"))
+            .build();
     }
 
     private static Chapter getPublishingContextChapter() {
         return new Chapter.Builder()
-                .withPartOf(SOME_URI)
-                .build();
+            .withPartOf(SOME_URI)
+            .build();
     }
 
     private static BasicContext getPublishingContextDegree() throws InvalidIsbnException,
-            InvalidUnconfirmedSeriesException {
+                                                                    InvalidUnconfirmedSeriesException {
         return new Degree.Builder()
-                .withIsbnList(List.of("9780201309515"))
-                .withPublisher(new UnconfirmedPublisher("My publisher dot com"))
-                .withSeriesNumber("123")
-                .withSeries(new UnconfirmedSeries("Explorations in ego"))
-                .build();
+            .withIsbnList(List.of("9780201309515"))
+            .withPublisher(new UnconfirmedPublisher("My publisher dot com"))
+            .withSeriesNumber("123")
+            .withSeries(UnconfirmedSeries.fromTitle("Explorations in ego"))
+            .build();
     }
 
     private static Report getPublishingContextReport() throws InvalidIssnException, InvalidIsbnException,
-            InvalidUnconfirmedSeriesException {
+                                                              InvalidUnconfirmedSeriesException {
         return new Report.Builder()
-                .withIsbnList(List.of("9780201309515"))
-                .withPublisher(new UnconfirmedPublisher("Hello cheesy world of anaemic flavours publishing"))
-                .withSeries(new UnconfirmedSeries("Strøsand and Muck in context"))
-                .withSeriesNumber("221")
-                .withOnlineIssn("1111-1119")
-                .withPrintIssn("2222-2227")
-                .build();
+            .withIsbnList(List.of("9780201309515"))
+            .withPublisher(new UnconfirmedPublisher("Hello cheesy world of anaemic flavours publishing"))
+            .withSeries(UnconfirmedSeries.fromTitle("Strøsand and Muck in context"))
+            .withSeriesNumber("221")
+            .withOnlineIssn("1111-1119")
+            .withPrintIssn("2222-2227")
+            .build();
     }
 
     private static String randomWord() {
         return FAKER.lorem().word() + FAKER.lorem().word();
     }
+
+    private static String random7DigitNumber() {
```

Comment:
Can we extract the ISSN generation code to a separate helper class?

Reply:
Moved to a different class